pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HdfsSegmentFetcher.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.segment.fetcher;
+
+import com.google.common.base.Strings;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.RETRY;
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.RETRY_WAITIME_MS;
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.HdfsSegmentFetcher.PRINCIPLE;
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.HdfsSegmentFetcher.KEYTAB;
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.HdfsSegmentFetcher.HADOOP_CONF_PATH;
+
+public class HdfsSegmentFetcher implements SegmentFetcher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HdfsSegmentFetcher.class);
+  private FileSystem hadoopFS = null;
+  private int retryCount = 3;
+  private int retryWaitMs = 100;
+
+  @Override
+  public void init(Map<String, String> configs) {
+    String hadoopConfPath = configs.get(HADOOP_CONF_PATH);
+    try {
+      if (configs.containsKey(RETRY)) {
+        retryCount = Integer.valueOf(configs.get(RETRY));
```

Comment:
Can you add a try/catch block and a log if it fails to parse? (eg. `Failed to parse value RETRY, using default value of retryCount`)

Reply:
changed the whole segment fetcher initialization argument to configuration to facilitate default/exception handling. The old way of passing in Map<String, String> is unnecessary and cause a lot more redundant codes. Also refactor other fetcher and updated tests